native_client/ctcdecode/setup.py

```diff
@@ -0,0 +1,147 @@
+    except AttributeError:
+        numpy_include = numpy.get_numpy_include()
+except ImportError:
+    numpy_include = ''
+    assert 'NUMPY_INCLUDE' in os.environ
+
+numpy_include = os.getenv('NUMPY_INCLUDE', numpy_include)
+numpy_min_ver = os.getenv('NUMPY_DEP_VERSION', '')
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--num_processes",
+    default=1,
+    type=int,
+    help="Number of cpu processes to build package. (default: %(default)d)")
+args = parser.parse_known_args()
+
+# reconstruct sys.argv to pass to setup below
+sys.argv = [sys.argv[0]] + args[1]
+
+def read(fname):
+    return open(os.path.join(os.path.dirname(__file__), fname)).read()
+
+project_version = read('../../VERSION').strip()
+
+# monkey-patch for parallel compilation
+# See: https://stackoverflow.com/a/13176803
+def parallelCCompile(self,
+                     sources,
+                     output_dir=None,
+                     macros=None,
+                     include_dirs=None,
+                     debug=0,
+                     extra_preargs=None,
+                     extra_postargs=None,
+                     depends=None):
+    # those lines are copied from distutils.ccompiler.CCompiler directly
+    macros, objects, extra_postargs, pp_opts, build = self._setup_compile(
+        output_dir, macros, include_dirs, sources, depends, extra_postargs)
+    cc_args = self._get_cc_args(pp_opts, debug, extra_preargs)
+
+    # parallel code
+    def _single_compile(obj):
+        try:
+            src, ext = build[obj]
+        except KeyError:
+            return
+        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+
+    # convert to list, imap is evaluated on-demand
+    thread_pool = multiprocessing.pool.ThreadPool(args[0].num_processes)
+    list(thread_pool.imap(_single_compile, objects))
+    return objects
+
+
+def compile_test(header, library):
+    dummy_path = os.path.join(os.path.dirname(__file__), "dummy")
+    command = "bash -c \"g++ -include " + header \
+                + " -l" + library + " -x c++ - <<<'int main() {}' -o " \
+                + dummy_path + " >/dev/null 2>/dev/null && rm " \
+                + dummy_path + " 2>/dev/null\""
+    return os.system(command) == 0
+
+
+# hack compile to support parallel compiling
+distutils.ccompiler.CCompiler.compile = parallelCCompile
+
+FILES = glob.glob('../kenlm/util/*.cc') \
+        + glob.glob('../kenlm/lm/*.cc') \
+        + glob.glob('../kenlm/util/double-conversion/*.cc')
+
+FILES += glob.glob('third_party/openfst-1.6.7/src/lib/*.cc')
+
+FILES = [
+    fn for fn in FILES
+    if not (fn.endswith('main.cc') or fn.endswith('test.cc') or fn.endswith(
+        'unittest.cc'))
+]
+
+LIBS = ['stdc++']
+if platform.system() != 'Darwin':
+    LIBS.append('rt')
+
+ARGS = ['-O3', '-DNDEBUG', '-DKENLM_MAX_ORDER=6', '-std=c++11',
+        '-Wno-unused-local-typedef', '-Wno-sign-compare']
+
+if compile_test('zlib.h', 'z'):
```

Comment:
I'm not really sure if it's a good idea, that feels like an easy way to forget things on different platforms and end up with strange behaviors. Shouldn't we just enforce the use of one of those? zlib, bz2 or lzma ?

Reply:
Yes, when removing the old CTC code I realized this doesn't match our build configuration inside libdeepspeech.so, so I'll just remove these tests and flags.

Reply:
ok